config/configcompression/compressiontype.go

```diff
@@ -24,18 +36,45 @@ func (ct *Type) IsCompressed() bool {
 	return *ct != typeEmpty && *ct != typeNone
 }
 
-func (ct *Type) UnmarshalText(in []byte) error {
-	typ := Type(in)
-	if typ == TypeGzip ||
-		typ == TypeZlib ||
-		typ == TypeDeflate ||
-		typ == TypeSnappy ||
-		typ == TypeZstd ||
-		typ == typeNone ||
-		typ == typeEmpty {
-		*ct = typ
+func (ct *TypeWithLevel) UnmarshalText(in []byte) error {
+	var compressionTyp Type
+	var level int
+	var err error
+	parts := strings.Split(string(in), "/")
+	compressionTyp = Type(parts[0])
+	level = zlib.DefaultCompression
+	if len(parts) > 1 {
+		level, err = strconv.Atoi(parts[1])
+		if err != nil {
+			return fmt.Errorf("invalid compression level: %q", parts[1])
+		}
+		if compressionTyp == TypeSnappy ||
+			compressionTyp == typeNone ||
+			compressionTyp == typeEmpty {
+			return fmt.Errorf("compression level is not supported for %q", compressionTyp)
+		}
+	}
+	ct.Level = Level(level)
+	if (compressionTyp == TypeGzip && isValidLevel(level)) ||
+		(compressionTyp == TypeZlib && isValidLevel(level)) ||
+		(compressionTyp == TypeDeflate && isValidLevel(level)) ||
+		compressionTyp == TypeSnappy ||
+		compressionTyp == TypeZstd ||
+		compressionTyp == typeNone ||
+		compressionTyp == typeEmpty {
```

Comment:
Let's fail if the level is provided for types that don't support it

Reply:
yes this is handled in the prior condition - https://github.com/open-telemetry/opentelemetry-collector/pull/11084/files#diff-ae510a297dd561956c65c1e375f12870c19e281c486b51017e3280e0d3de08f2R51

Reply:
Here's the test for it - https://github.com/open-telemetry/opentelemetry-collector/pull/11084/files#diff-fc75b0e352923bac06c0d1b09268c82efe3ad8523c0107926c5e919d4ab9dceeR78